src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -571,8 +582,39 @@ def getMSOutputTransferInfo(self, wflow):
             msg = "General exception while fetching TransferInfo from MSOutput for %s. "
             msg += "Error: %s"
             self.logger.exception(msg, wflow['RequestName'], str(ex))
+
+        # Set Transfer status - information fetched from MSOutput only
         if transferInfo is not None and transferInfo['TransferStatus'] == 'done':
             wflow['TransferDone'] = True
+
+        # Set Tape rules status - information fetched from Rucio (tape rule ids from MSOutput)
+        if transferInfo is not None and transferInfo['OutputMap']:
+            tapeRulesStatusList = []
+            # For setting 'TransferTape' = True we require either no tape rules for the
+            # workflow have been created or all existing tape rules to be in status 'OK',
+            # so every empty TapeRuleID we consider as completed.
+            for ruleId in [mapRecord['TapeRuleID'] for mapRecord in transferInfo['OutputMap']]:
+                if not ruleId.strip():
+                    tapeRulesStatusList.append(True)
+                else:
+                    rucioRule = self.rucio.getRule(ruleId)
+                    if rucioRule:
+                        if rucioRule['state'] == 'OK':
+                            tapeRulesStatusList.append(True)
+                            msg = "Tape rule: %s in final state: %s for workflow: %s"
+                            self.logger.info(msg, ruleId, rucioRule['state'], wflow['RequestName'])
+                        else:
+                            tapeRulesStatusList.append(False)
+                            msg = "Tape rule: %s in non final state: %s for workflow: %s"
+                            self.logger.info(msg, ruleId, rucioRule['state'], wflow['RequestName'])
+                    else:
+                        tapeRulesStatusList.append(False)
+                        msg = "Tape rule: %s not found for workflow: %s "
+                        msg += "Possible server side error or wrong Rucio server configured for MSRuleCleaner."
```

Comment:
I would suggest to remove `or wrong Rucio server configured`. This should never be the case, unless it's in our private nodes and we are messing with things.

Reply:
Done